### PR TITLE
Sysfs debug hw

### PR DESCRIPTION
**Pull request by virkha**
Kernel debug: sysfs solution

### CODE REVIEW

File: 05_debug/debug_module.c [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+ssize_t sysfs_store(struct kobject *kobj, struct kobj_attribute *attr,
+		const char *buf, size_t count)
+{
+	int i = 0;
+
+	num_calls_write++;
+	memset(buffer, 0, sizeof(buffer));
+	strncpy(buffer, buf, count);
+	buffer[count] = '\0';
```
Comment (ekovalyov) [inline 05_debug/debug_module.c:93]: What is the size of buffer?

File: 05_debug/debug_module.c [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	return sprintf(buf, buffer);
+}
+
+ssize_t sysfs_store(struct kobject *kobj, struct kobj_attribute *attr,
+		const char *buf, size_t count)
+{
+	int i = 0;
+
+	num_calls_write++;
+	memset(buffer, 0, sizeof(buffer));
```
Comment (ekovalyov) [inline 05_debug/debug_module.c:91]: is this memset required?

Comment (virkha) [reply 05_debug/debug_module.c:91]: memset cleans the buffer, delete previous passed value, before store new one.
If don't clean buffer, we can observe unexpected fragments behind storing value.

File: 05_debug/debug_module.c (hunk 1 above)
Comment (virkha) [reply 05_debug/debug_module.c:93]: size of buffer is PAGE_SIZE.
do you mean add extra check if count < PAGE_SIZE before fill buffer?

**Review by virkha — COMMENTED** [review]
issues commented

File: 05_debug/debug_module.c (hunk 1 above)
Comment (ekovalyov) [reply 05_debug/debug_module.c:93]: It is good idea to check all possible input values.

**Review by ekovalyov — APPROVED** [review]

**Pull request merged**